de.metas.business/src/main/java/de/metas/purchasing/api/impl/BPartnerProductDAO.java

```diff
@@ -271,10 +274,41 @@ private static ProductExclude toProductExclude(final I_C_BPartner_Product bpartn
 				.build();
 	}
 
+
 	@Override
 	public Optional<ProductExclude> getExcludedFromSaleToCustomer(final int productId, final int partnerId)
 	{
-		final I_C_BPartner_Product bpartnerProduct = Services.get(IQueryBL.class)
+		final Optional<ProductExclude> productExcluded = getExcludedProductFromSaleToCustomer(productId, partnerId);
+		final Optional<ProductExclude> manufacturerExcluded = getBannedManufacturerFromSaleToCustomer(productId, partnerId);
+
+		if (productExcluded.isPresent() || manufacturerExcluded.isPresent())
+		{
+			final ProductExcludeBuilder builder = ProductExclude.builder()
+					.bpartnerId(partnerId)
```

Comment:
use BPartnerId and ProductId objects instead of ints. Use them as method params, change the ProductExclude etc